modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -173,24 +176,73 @@ private boolean usesRetrofit2Library() {
                 List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
                 for (CodegenOperation operation : ops) {
                     if (operation.hasConsumes == Boolean.TRUE) {
-                        Map<String, String> firstType = operation.consumes.get(0);
-                        if (firstType != null) {
-                            if ("multipart/form-data".equals(firstType.get("mediaType"))) {
-                                operation.isMultipart = Boolean.TRUE;
-                            }
+
+                        if ( isMultipartType(operation.consumes) ) { 
+                            operation.isMultipart = Boolean.TRUE;
+                        }	
+                        else {
+                            operation.prioritizedContentTypes = prioritizeContentTypes(operation.consumes);
                         }
                     }
                     if (operation.returnType == null) {
                         operation.returnType = "Void";
                     }
                     if (usesRetrofit2Library() && StringUtils.isNotEmpty(operation.path) && operation.path.startsWith("/"))
-                    	operation.path = operation.path.substring(1);
+                        operation.path = operation.path.substring(1);
                 }
             }
         }
         return objs;
     }
 
+    /**
+     *  Prioritizes consumes mime-type list by moving json-vendor and json mime-types up front, but 
```

Comment:
Instead of moving json-vendor and json mime-types up front, what about maintaining the original order of the elements in `consumes`, `produces`? (currently the order is not maintained)